src/vstest.console/Processors/TestCaseFilterArgumentProcessor.cs

```diff
@@ -117,12 +117,23 @@ public TestCaseFilterArgumentExecutor(CommandLineOptions options)
         /// <param name="argument">Argument that was provided with the command.</param>
         public void Initialize(string argument)
         {
-            if (string.IsNullOrWhiteSpace(argument))
+            var defaultFilter = this.commandLineOptions.TestCaseFilterValue;
+            var hasDefaultFilter = !string.IsNullOrWhiteSpace(defaultFilter);
+            
+            if (!hasDefaultFilter && string.IsNullOrWhiteSpace(argument))
             {
                 throw new CommandLineException(string.Format(CultureInfo.CurrentUICulture, CommandLineResources.TestCaseFilterValueRequired));
             }
 
-            this.commandLineOptions.TestCaseFilterValue = argument;
+            if (!hasDefaultFilter)
```

Comment:
nit: This could be re-written as a conditional expression (see below). To me it appears cleaner and more compact this way.

```
this.commandLineOptions.TestCaseFilterValue = (!hasDefaultFilter)
    ? argument
    : $"({defaultFilter})&({argument})";
```

Reply:
Style preference, does not look more clean to me, especially since not all ifs in this block are ternary 🙂 Won't change